app/controllers/api/v1/have_read_books_controller.rb

```diff
@@ -0,0 +1,50 @@
+class Api::V1::HaveReadBooksController < ApplicationController
+  def create
+    have_read_book = HaveReadBook.new(ajax_params)
+    have_read_book.user = current_user
+    have_read_book.build_date(date_params)
+
+    if have_read_book.save
+      to_read_book = ToReadBook.find_by(user: current_user, book_id: ajax_params[:book_id])
+      current_user.update_ranks(ajax_params[:book_id], to_read_book.rank, nil)
+      to_read_book&.destroy
+      trb_books = current_user.to_read_books.where.not(rank: nil).map { |trb| {trb_book_id: trb.book_id, rank: trb.rank} }
+
+      render json: {
+        book_id: "#{ajax_params[:book_id]}",
+        trb_books: trb_books,
+        status: :created
+      }
+    else
+      render json: :nothing, status: :not_found
+    end
+  end
+
+  def update
+    have_read_book = HaveReadBook.find(params[:id])
+    have_read_book.build_date(date_params)
+    have_read_book.rating = ajax_params[:rating]
+    have_read_book.note = ajax_params[:note]
+
+    if have_read_book.save
+      render json: {
+        book_id: "#{ajax_params[:book_id]}",
```

Comment:
Do these need interpolation or could you just point to the variables not wrapped in strings?